pyintegration/deephaven2/table.py

```diff
@@ -88,6 +40,15 @@ def __repr__(self):
         repr_str = repr_str[:115] + "...}>" if len(repr_str) > 120 else repr_str
         return repr_str
 
+    def __eq__(self, table: Table) -> bool:
+        try:
+            if _JTableTools.diff(self.j_table, table.j_table, 1):
+                return False
+            else:
+                return True
+        except Exception as e:
+            raise DHError(e, "table equality test failed.") from e
```

Comment:
It's not actually indirect at all. It's a chunk-oriented `equals` tool that also describes the differences between the two tables, because most of the time it's not helpful to just know that they're unequal. I would support adding an explicit `equals` method that wraps diff, if we want.
Unit tests generally use `io.deephaven.engine.table.impl.TstUtils#assertTableEquals(io.deephaven.engine.table.Table, io.deephaven.engine.table.Table, io.deephaven.engine.util.TableDiff.DiffItems...)` which is implemented in terms of diff.